scripts/install_xcpng.py

```diff
@@ -62,24 +62,58 @@ def generate_answerfile(directory, installer, hostname_or_ip, type, hdd):
         elif type == 'upgrade':
             answerfile.write(f"""<?xml version="1.0"?>
 <installation mode="upgrade">
-    <existing-installation>sda</existing-installation>
+    <existing-installation>{hdd}</existing-installation>
     <source type="url">{installer}</source>
     <script stage="filesystem-populated" type="url">
         http://{pxe}/configs/presets/scripts/filesystem-populated.py
     </script>
 </installation>
         """)
+        elif type == 'restore':
+            answerfile.write(f"""<?xml version="1.0"?>
+<restore>
+    <existing-installation>{hdd}</existing-installation>
+    <source type="url">{installer}</source>
+    <script stage="filesystem-populated" type="url">
+        http://{pxe}/configs/presets/scripts/filesystem-populated.py
+    </script>
+</restore>
+        """)
+        else:
+            raise Exception(f"Unknown type `{type}`")
+
+def generate_pxelinux(directory, installer, type, mac_address):
+    pxe = pxe_address()
+    with open(f'{directory}/pxelinux.cfg', 'w') as pxelinux:
+        if type == 'restore':
+            pxelinux.write(
+                f"""DEFAULT menu.c32
+PROMPT 0
+MENU TITLE Main Menu
+TIMEOUT 50
+ONTIMEOUT custom
+
+LABEL custom
+    MENU LABEL Custom XCP-ng installation
+    KERNEL mboot.c32
+    APPEND {installer}/boot/xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga \
+--- {installer}/boot/vmlinuz xencons=hvc console=hvc0 console=tty0 \
+rt_answerfile=http://{pxe}/configs/custom/{mac_address}/answerfile.xml install \
+--- {installer}install.img"""
+            )
```

Comment:
As I told you earlier, using `pxelinux.cfg` was to be done manually, as a way to test the `rt_answerfile` option before. It was not supposed to end up in this script. I also said after the manual tests we would improve `bootconfig.php` so that it would support a `rt_answerfile` option in `boot.conf`.

By the way pxelinux is not used in UEFI mode so the script probably doesn't work with UEFI XCP-ng.

In the future, maybe we'll add support for writing `pxelinux.cfg` and `grub.cfg` files in this script (so let's save this snipped somewhere), but for now `boot.conf` is the way.